packages/auth/amplify_auth_cognito_dart/lib/src/model/session/cognito_auth_session.dart

```diff
@@ -2,52 +2,48 @@
 // SPDX-License-Identifier: Apache-2.0
 
 import 'package:amplify_auth_cognito_dart/amplify_auth_cognito_dart.dart';
+import 'package:amplify_auth_cognito_dart/src/model/auth_result.dart';
 import 'package:amplify_core/amplify_core.dart';
 
-part 'cognito_auth_session.g.dart';
-
 /// {@template amplify_auth_cognito.model.cognito_auth_session}
 /// The current Cognito auth session, with AWS credentials and User Pool tokens
 /// of the active user.
 /// {@endtemplate}
-@zAmplifySerializable
 class CognitoAuthSession extends AuthSession
     with
         AWSEquatable<CognitoAuthSession>,
         AWSSerializable<Map<String, Object?>> {
   /// {@macro amplify_auth_cognito.model.cognito_auth_session}
   const CognitoAuthSession({
     required super.isSignedIn,
-    this.credentials,
-    this.userPoolTokens,
-    this.userSub,
-    this.identityId,
+    required this.userPoolTokensResult,
+    required this.userSubResult,
+    required this.credentialsResult,
+    required this.identityIdResult,
   });
 
-  /// {@macro amplify_auth_cognito.model.cognito_auth_session}
-  factory CognitoAuthSession.fromJson(Map<String, Object?> json) =>
-      _$CognitoAuthSessionFromJson(json);
-
-  /// The AWS credentials.
-  final AWSCredentials? credentials;
-
   /// The User Pool tokens.
-  final CognitoUserPoolTokens? userPoolTokens;
```

Comment:
Can we leave these, and mark them as deprecated to a make this relatively non-breaking?

```dart
@Deprecated('Use `userPoolTokensResult.value` instead')
CognitoUserPoolTokens? get userPoolTokens => userPoolTokensResult.valueOrNull;
```